### PR TITLE
Updated Lazarus for macOS builds to 2.0.8

### LINKED ISSUE #1261: Multi download selection malfunction
1. Operating system : Mac Mojave
2. Transmission Deamon : 2.94-16
3. Transmission Remote GUI : 5.18.0 Build 3fb1d33

When I select download mission (not sequential) always use command + left click, 
but somehow it doesn't work at 5.18.0. Sequential select with shift + left click is worked.

### LINKED ISSUE #1268: Dark mode on macOS
Text in the app is not visible.
Transmission Remote GUI 5.18.0
macOS Catalina 10.15.1 > Preferences > General > Appearance (Dark) > Highlight color (Graphite)

### LINKED ISSUE #1272: Up/Down Limits not saving 5.18.0 Mac
It seems that when I set and save the bandwidth limits for items this does not properly save or register under 5.18.0 on OSX and keeps that at whatever the global is, or the last value that was set. This does work as expected in 5.17.0. 

1. Operating system (OS X Mojave 10.14.6)
2. Transmission Deamon (Transmission 2.92/Linux CentOS)
3. Transmission Remote GUI (5.18.0)

### LINKED ISSUE #1277: Cannot manually type the port on the app (version 5.18.0 on macOS  Mojave)
as above.

Steps to reproduce

1. Go to manage connections
2. Change the port by manually typing it
3. Press ok to save changes 
4. Go back to manage connections

If i use the up/down arrows, it works fine.

Edit: temporary fix is to manually edit the ini file

### LINKED ISSUE #1289: UI Issues on Mac
I have the 5.18.0 version, but it has some issues with the interface. Testing on Mac Catalina dark mode. The selection bars are the same color as font colors (white) and because of this I'm unable to read the selected row. Looking at the image in project description, it probably should be green.

screenshot

### PR DESCRIPTION
**Pull request by someperson**
- Updated Lazarus for macOS builds to 2.0.8
- Updated Xcode for macOS builds to 11.3

This fixes #1272 and fixes #1277, and also partially fixes #1268 (switching between dark and light mode while transgui is running still isn't handled properly). May also address other issues.

EDIT: Also added a bugfix in here that fixes #1261

### CODE REVIEW

File: setup/macosx/install_deps.sh [hunk 1]
```diff
@@ -3,29 +3,23 @@
...
-  wget "https://downloads.sourceforge.net/project/lazarus/Lazarus%20Mac%20OS%20X%20i386/Lazarus%20${lazarus_ver}/$fpc.dmg?r=&ts=$(date +%s)${mirror_string-}" -O "$fpc.dmg"
-  hdiutil attach -quiet "$fpc.dmg"
-  pkgpath="$(hdiutil attach "$fpc.dmg" | command awk "/Apple_HFS/ { print \$3 }")"
-  sudo installer -pkg "$pkgpath/$fpc.pkg" -target /
-  hdiutil unmount "$pkgpath"
-  rm "$fpc.dmg"
+  wget "https://downloads.sourceforge.net/project/lazarus/Lazarus%20macOS%20x86-64/Lazarus%20${lazarus_ver}/$fpc.pkg?r=&ts=$(date +%s)${mirror_string-}" -O "fpc.pkg"
+  sudo ln -s /usr/local/lib/fpc/3.0.4/ppcx64 /usr/local/bin/ppcx64
+  sudo installer -pkg "fpc.pkg" -target /
+  rm "fpc.pkg"
```
Comment (PeterDaveHello) [inline setup/macosx/install_deps.sh:18]: broken here

File: setup/macosx/install_deps.sh [hunk 2]
```diff
@@ -3,29 +3,23 @@
...
-  wget "https://downloads.sourceforge.net/project/lazarus/Lazarus%20Mac%20OS%20X%20i386/Lazarus%20${lazarus_ver}/$lazarus.dmg?r=&ts=$(date +%s)${mirror_string-}" -O "$lazarus.dmg"
-  hdiutil attach -quiet "$lazarus.dmg"
-  pkgpath="$(hdiutil attach "$lazarus.dmg" | command awk "/Apple_HFS/ { print \$3 }")"
-  sudo installer -pkg "$pkgpath/lazarus.pkg" -target /
-  hdiutil unmount "$pkgpath"
-  rm "$lazarus.dmg"
-  lazbuild --build-ide= --compiler=fpc --cpu=x86_64 --widgetset=cocoa
+  wget "https://downloads.sourceforge.net/project/lazarus/Lazarus%20macOS%20x86-64/Lazarus%20${lazarus_ver}/$lazarus.pkg?r=&ts=$(date +%s)${mirror_string-}" -O "lazarus.pkg"
+  sudo installer -pkg "lazarus.pkg" -target /
+  rm "lazarus.pkg"
```
Comment (PeterDaveHello) [inline setup/macosx/install_deps.sh:24]: also broken here

File: setup/macosx/install_deps.sh (hunk 1 above)
Comment (someperson) [reply setup/macosx/install_deps.sh:18]: How is this broken? Seems to work on Travis CI, and I even stood up a new macOS VM and it worked there. What error(s) are you running into?

**Comment by someperson** [comment]
Added a change to fix #1261

**Comment by someperson** [comment]
Could you please clarify how this is broken? I'm not running into any problems with it, even with a freshly-installed macOS in a VM.

**Comment by fabrykowski** [comment]
Any news?

**Comment by flying-sausages** [comment]
> Could you please clarify how this is broken? I'm not running into any problems with it, even with a freshly-installed macOS in a VM.

I mean, the logs suggest some debian package failed to install on the docker VM. This could be an issue in the CI itself most likely. Looks like It's in `/debian-ubuntu-install_deps.sh`. This might have been fixed in master in the meantime, any chance you could merge that in and see what Travis has to say?

File: setup/macosx/install_deps.sh (hunk 1 above)
Comment (PeterDaveHello) [reply setup/macosx/install_deps.sh:18]: Sorry, I might comment on the wrong line.

**Comment by PeterDaveHello** [comment]
@someperson sorry for the very late response, somehow I missed it, will check it now.

**Comment by PeterDaveHello** [comment]
Currently, I'm trying to apply open-source credits for this project on Travis CI, so that I can verify the process on macOS ;)

**Review by PeterDaveHello — APPROVED** [review]
Thanks @someperson

**Comment by jordanade** [comment]
Any update here? Would love to see a new release that fixes #1261

**Comment by PeterDaveHello** [comment]
Sorry that I don't have enough bandwidth to draft a new release recently, will see if I can make some time in the next few months...

**Pull request merged**